frontend/src/app/rating/rating.page.ts

```diff
@@ -4,38 +4,280 @@
+          (To be more precise: It might fire even more often than that.)
+          So, we can use it to update our progress bar.
+          Of course, there is still room for performance improvement (since we will call the following method really often).
+          But for now, it should be fine.
+        */
+        this.updateProgessStatus();
+      }
+    );
+    this.subscriptions.push(formStatusChangesSubscriptions);
+  }
+
+  /**
+   * Unsubscribe from all of our Subscriptions.
+   */
+  ngOnDestroy(): void {
+    for (const subscription of this.subscriptions) {
+      subscription.unsubscribe();
+    }
+  }
+
+  /**
+   * Handle a Step as provided by the Router.
+   *
+   * The string provided by the Router could contain anything.
+   * So: If the Step is not known, we simply show the Applicant rating step.
+   *
+   * (When navigating to the URL without providing a query parameter, we will also redirect to the Applicant rating step).
+   *
+   * @param step The "step" query parameter we got from the Router.
+   */
+  private handleStep(step: unknown): void {
+    // tslint:disable-next-line: no-any
+    if (hrRatingStepArr.includes(step as any)) {
+      this.setCurrentStep(step as hrRatingStep);
+      return;
+    }
+
+    if (step) {
+      console.warn('The following Step is unknown:', step);
+    }
+
+    // A fallback: If a step was requested that we don't know, we simply show the Applicant rating page.
+    this.navigateToStep(hrRatingStep.ApplicantRating);
+  }
+
+  /**
+   * Set the currentStep and currentStepIndex property.
+   *
+   * Usually, if you want to change the Step, you should use the Router.
+   */
+  private setCurrentStep(step: hrRatingStep): void {
+    this.currentStep = step;
+    this.currentStepIndex = hrRatingStepArr.indexOf(step);
+  }
+
+  /**
+   * Update the "step" query parameter.
+   * You can use this to navigate between the different form steps!
+   */
+  navigateToStep(step: hrRatingStep): void {
+    this.router.navigate([], {
+      relativeTo: this.activatedRoute,
+      queryParams: {
+        step,
+      },
+    });
+  }
+
+  /**
+   * In this method navigation to next step is handled.
+   */
+  navigateToNextStep(): void {
+    if (this.currentStepIndex < hrRatingStepArr.length - 1) {
+      const step = hrRatingStepArr[this.currentStepIndex + 1];
+      this.navigateToStep(step);
+
+      return;
+    }
+
+    console.warn('There is no next step.');
+  }
   /**
    * In this method navigation to home page is handled.
    */
   goToHomePage(): void {
     this.navController.navigateBack('/home');
   }
+  /**
+   * In this method popup is displayed to confirm finalize the applicant information
+   */
+  async finalize(): Promise<void> {
+    const alert = await this.alertController.create({
+      header: 'Finalize',
+      message:
+        'Do you really want to finalize this form? it wont be possible to edit later.',
```

Comment:
There is a typo: "wont" instead of "won't" :) And the "it" should be uppercase.

Reply:
could not figure out how to make wont to won't. I tried with escape sequence but couldn't do it as lint fails.

Reply:
Can't you just use `const str = 'this won\'t be a problem'`? Or have we made the linter too strict? :D